src/host/renderData.cpp

```diff
@@ -79,6 +79,16 @@ std::vector<Viewport> RenderData::GetSelectionRects() noexcept
     return result;
 }
 
+// Method Description:
+// - Retrieves one rectangle per line describing the area of the viewport
+//   that should be highlighted in some way to represent a user-interactive selection
+// Return Value:
+// - Vector of Viewports describing the area selected
+std::vector<Viewport> RenderData::GetSearchSelectionRects() noexcept
+{
+    return {};

```

Comment:
Conhost performance question: it's cheap to return an empty vector, right?

Gut check... it looks like it is "cheap enough", at least on x64:

```c++
#include <vector>
std::vector<int> foo() {
    return {};
}
```
```nasm
__$ReturnUdt$ = 32
std::vector<int,std::allocator<int> > foo(void) PROC  ; foo, COMDAT
$LN13:
        sub     rsp, 24
        xor     eax, eax
        mov     QWORD PTR [rcx], rax
        mov     QWORD PTR [rcx+8], rax
        mov     QWORD PTR [rcx+16], rax
        mov     rax, rcx
        add     rsp, 24
        ret     0
std::vector<int,std::allocator<int> > foo(void) ENDP  ; foo
```

Reply:
This is way beyond me,  ChatGpt seems to think it was optimized since it was using Return Value Optimization to move the return instead of copying it.